### PR TITLE
[NUI] Add GlWindow

### PR DESCRIPTION
**Pull request by dkdk-ryu**
GlWindow allows you to render scenes with OpenGL ES.

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by hyunjushin** [comment]
please share GlWindow sample application code.

File: src/Tizen.NUI/src/public/GlWindow.cs [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+        public delegate void GlTerminateType();
+
+        /// <summary>
+        /// Registers a GL callback function for application.
+        /// </summary>
+        /// <param name="glInit">The callback function for application initialize</param>
+        /// <param name="glRenderFrame">The callback function to render to the frame</param>
+        /// <param name="glTerminate">The callback function to clean-up application GL resource</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void RegisterGlCallback( GlInitializeType glInit, GlRenderFrameType glRenderFrame, GlTerminateType glTerminate )
```
Comment (sparrow74) [inline src/Tizen.NUI/src/public/GlWindow.cs:481]: Modify this function to use local value. please.
Refer my repository of GLWindow_v1.

File: src/Tizen.NUI/src/public/GlWindowEvent.cs [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+ */
+
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI
+{
+    public partial class GlWindow
```
Comment (sparrow74) [inline src/Tizen.NUI/src/public/GlWindowEvent.cs:25]: Please, add [EditorBrowsable(EditorBrowsableState.Never)].

**Review by hyunjushin — APPROVED** [review]

**Review by hyunjushin — CHANGES_REQUESTED** [review]
check warning messages

**Comment by WonyoungChoi** [comment]
I think `GL` prefix is better than `Gl`.  It is generally used in capital letters as a proper noun.

**Comment by idkiller** [comment]
When implementing property getter / setter or add / remove of event, you can see code that initializes by null check in most situations.
If you use Lazy<T>, you can keep the code more concise and nice, so it would be nice to consider using it.

File: src/Tizen.NUI/src/public/GlWindow.cs [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+        {
+            init1 = glInit;
+            init2 = new HandleRef(this, Marshal.GetFunctionPointerForDelegate<Delegate>(init1));
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/GlWindow.cs:492]: could you do naming the values as meaningful one?
also could you check this link?

https://docs.microsoft.com/ko-kr/dotnet/api/system.gc.keepalive?view=netcore-3.1

if these variables are for blocking garbage collected, then you can use KeepAlive maybe...

File: src/Tizen.NUI/src/public/GlWindowEvent.cs [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+            if (VisibilityChangedEventHandler != null)
+            {
+                VisibilityChangedEventHandler.Invoke(this, e);
+            }
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void GlVisibilityChangedEventCallbackType(IntPtr window, bool visibility);
+        private GlVisibilityChangedEventCallbackType GlVisibilityChangedEventCallback;
+        private event EventHandler<VisibilityChangedArgs> VisibilityChangedEventHandler;
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/GlWindowEvent.cs:406]: this is trivial one. but could you decide whether underscore( _ ) is used for private variable or not?
some has underscore but this one doesn't have it... ^^;

**Comment by dkdk-ryu** [comment]
> I think `GL` prefix is better than `Gl`. It is generally used in capital letters as a proper noun.

I changed them.

**Comment by dongsug-song** [comment]
@hyunjushin 
@idkiller 

Hello, 
Actually, this patch needs be merged urgently. 
(and these API are all hidden API)
I talked with Mr. dkdk-ryu and we will do refactoring later soon.
(about using Lazy<T>, handling exceptions and etc...)
so for now, I want you to accept this to be merged.
Thank you.

**Review by dongsug-song — APPROVED** [review]

**Review by hyunjushin — APPROVED** [review]

**Pull request merged**